src/main/java/org/folio/circulation/infrastructure/storage/notes/NoteTypesRepository.java

```diff
@@ -0,0 +1,32 @@
+package org.folio.circulation.infrastructure.storage.notes;
+
+import static org.folio.circulation.support.ResultBinding.flatMapResult;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.MultipleRecords;
+import org.folio.circulation.domain.NoteType;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.http.client.CqlQuery;
+import org.folio.circulation.support.http.client.PageLimit;
+import org.folio.circulation.support.http.client.Response;
+
+public class NoteTypesRepository {
+  private final CollectionResourceClient noteTypesClient;
+
+  public NoteTypesRepository(Clients clients) {
+    noteTypesClient = clients.noteTypesClient();
+  }
+
+  public CompletableFuture<Result<MultipleRecords<NoteType>>> findByName(String name) {
```

Comment:
Out of topic: @marcjohnson-kint, @rladdusaw do we have an agreement whether we should favor using `MultipleRecords` or a java `Collection`?

I saw both approaches, and some times we have a method that does the same logic as I looking for, but for a different collection type.

Reply:
I think that depends upon the usage. 

I increasingly think there is an anti-pattern of usage with `MultipleRecords` where it is used for any collection, and in those circumstances the overall record count is often misleading.

What we might need is some specific methods that return `MultipleRecords` or a new class (maybe `PagedRecords`) specifically for when there is an intention to pass much of the response onto the client to mod-circulation.

In other cases, a regular collection or similar can likely be used and would be more descriptive.